src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -337,6 +341,18 @@ public void OnException(Activity activity, object payload)
                 return;
             }
 
+            if (this.emitNewAttributes)
+            {
+#if NET8_0_OR_GREATER
+                // For net8.0 and above exception type can be found using HttpRequestError.
+                // https://learn.microsoft.com/dotnet/api/system.net.http.httprequesterror?view=net-8.0
+                var errorType = GetErrorType(exc);
+#else
+                var errorType = exc.GetType().FullName;
+#endif
```

Comment:
nit: I think this comment would be better in the `GetErrorType` method summary.
This If/Else could be moved into that method.